src/lib.rs

```diff
@@ -1,3 +1,22 @@
+#![warn(missing_docs)]
+//! # Motivation and Goals
+//!
+//! 1. Ergonomics is first priority. Being able to prototype quickly and code
```

Comment:
But we have `#![warn(missing_docs)]`. Should I remove that? (I've documented everything)